formula/src/main/java/com/instacart/formula/internal/FormulaManagerImpl.kt

```diff
@@ -31,47 +32,50 @@ internal class FormulaManagerImpl<Input, State, Output>(
     private var isValidationEnabled: Boolean = false
 
     private val actionManager: ActionManager = ActionManager(
+        manager = this,
         formulaType = type,
         inspector = inspector,
     )
 
+    var transitionID: Long = 0
     var terminated = false
 
+    fun hasTransitioned(id: Long): Boolean {
+        return transitionID != id
+    }
+
     fun handleTransitionResult(result: Transition.Result<State>) {
         if (terminated) {
             // State transitions are ignored, only side effects are passed up to be executed.
-            transitionListener.onTransitionResult(type, result, true)
+            delegate.onTransition(type, result, false)
```

Comment:
looking at this without digging in, seems suspicious that it went from true to false. Maybe we can add named arguments here since at least in GitHub, its hard to know what is what.

Reply:
Added named parameter for this